src/luascript.cpp

```diff
@@ -9159,9 +9162,37 @@ int LuaScriptInterface::luaPlayerSetBankBalance(lua_State* L)
 	return 1;
 }
 
+int LuaScriptInterface::luaPlayerHasStorage(lua_State* L)
+{
+	// player:hasStorage(key)
+	Player* player = getUserdata<Player>(L, 1);
+	if (!player) {
+		lua_pushnil(L);
+		return 1;
+	}
+
+	lua_pushboolean(L, player->storageMap.find(getNumber<uint32_t>(L, 2)) != player->storageMap.end());
+	return 1;
+}
+
+int LuaScriptInterface::luaPlayerClearStorage(lua_State* L)
```

Comment:
Totally unnecessary to declare `clearStorage` in sources, it can easily go in `data/lib/core/player.lua`
the name is also confusing, better `removeStorage` since that is what it actually does, remove it.